db/mongo/client.go

```diff
@@ -60,101 +61,71 @@ func (c *client) AddUser(user *domain.User) (string, error){
+    if err != nil {
+        return nil, err
+     }
+	fmt.Println(cur)
+	//filter changes end
+	// 	if queryName != ""{
+	// 		options := options.Find()
+	// 		options.SetLimit(int64(queryLimit))
+	// 		cur, err = c.dbc.Collection("users").Find(context.Background(),bson.M{"name" : queryName},options)
+	// 		fmt.Println("The name query is",queryName)
+	// 	} else{
+	// 		options := options.Find()
+	// 		options.SetLimit(int64(queryLimit))
+	// 		cur, err = c.dbc.Collection("users").Find(context.Background(),bson.D{},options)
+	// 	}
+	// if err != nil { log.Fatal(err) }
+	// defer cur.Close(context.Background())
 	resultArray:=[]*domain.User{}
-	for cur.Next(ctx) {
-		var result bson.D
+	for cur.Next(context.Background()) {
 		var result1 *domain.User	
 		err := cur.Decode(&result1)
 		if err != nil { log.Fatal(err) }
 		resultArray=append(resultArray,result1)
-		fmt.Println("The result is",result)
 	}
 	if err := cur.Err(); err != nil {
 		log.Fatal(err)
 	}
-	fmt.Println("The final result(List User) is",resultArray)
 	return resultArray,nil
 }
 func (c *client) DeleteUser(id string) error{
-	//Delete 
-	// ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
-	// defer cancel()
-	
-	// cur, err := c.dbc.Collection("users").Find(ctx, bson.M{"_id" : id})
-	// if err != nil { log.Fatal(err) }
-	// defer cur.Close(ctx)
-	// resultArray:=[]*domain.User{}
-	// for cur.Next(ctx) {
-	// 	// var result bson.D
-	// 	var result1 *domain.User	
-	// 	err := cur.Decode(&result1)
-	// 	if err != nil { log.Fatal(err) }
-	// 	resultArray=append(resultArray,result1)
-	// }
-	// if len(resultArray) == 0{
-	// 	return &domain.Error{Code:404,Message:"User doesn't exist"}
-	// }
 	c.dbc.Collection("users").DeleteOne(context.Background(), bson.M{"_id": id})
     return nil
 }
 
 func (c *client) ViewUser(id string) (*domain.User,error){
-	//View User 
-	// collection := c.dbc.Collection("users")
 	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
 	defer cancel()
-	//objID, err := primitive.ObjectIDFromHex(id)
-	// if err != nil {
-	// 	return nil,err
-	//   }
 	var userInfo domain.User
 	if err := c.dbc.Collection("users").FindOne(ctx, bson.M{"_id" : id}).Decode(&userInfo); err !=nil{
 		return nil,&domain.Error{Code:404,Message:"User doesn't exist"}
 	}
-	
-	// if err != nil { log.Fatal(err) }
-	// defer cur.Close(ctx)
-	// resultArray:=*domain.User
-	// for cur.Next(ctx) {
-	// 	var result bson.D
-	// 	var result1 *domain.User	
-	// 	err := cur.Decode(&result1)
-	// 	if err != nil { log.Fatal(err) }
-	// 	resultArray=append(resultArray,result1)
-	// 	fmt.Println("The viewed result is",result)
-	// }
-	// if err := cur.Err(); err != nil {
-	// 	log.Fatal(err)
-	// }
 	return &userInfo,nil
+}
+
+func applyFilter(filterMap map[string]interface{}) map[string]interface{} {
+    for k, v := range filterMap {
+        switch mval := v.(type) {
+        case []string:
```

Comment:
what is the purpose of this case?

Reply:
Sir currently it is not in use, but for suppose we want to search for multiple names at a time, then it can be used to get those records.

Reply:
yes, if interface is type of []string then it will treat as IN in db query.